data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetNewBlock.java

```diff
@@ -96,42 +114,40 @@ public GetNewBlock(final ValidatorDataProvider provider, final JsonProvider json
         @OpenApiResponse(status = RES_SERVICE_UNAVAILABLE, description = SERVICE_UNAVAILABLE)
       })
   @Override
-  public void handle(final Context ctx) throws Exception {
-    try {
-      final Map<String, List<String>> queryParamMap = ctx.queryParamMap();
-      final Map<String, String> pathParamMap = ctx.pathParamMap();
-
-      final UInt64 slot = UInt64.valueOf(pathParamMap.get(SLOT));
-      final BLSSignature randao = getParameterValueAsBLSSignature(queryParamMap, RANDAO_REVEAL);
-      final Optional<Bytes32> graffiti =
-          getParameterValueAsBytes32IfPresent(queryParamMap, GRAFFITI);
-      ctx.future(
-          provider
-              .getUnsignedBeaconBlockAtSlot(slot, randao, graffiti)
-              .thenApplyChecked(
-                  maybeBlock -> {
-                    if (maybeBlock.isEmpty()) {
-                      throw new ChainDataUnavailableException();
-                    }
-                    return produceResultString(maybeBlock.get());
-                  })
-              .exceptionallyCompose(error -> handleError(ctx, error)));
-    } catch (final IllegalArgumentException e) {
-      ctx.status(SC_BAD_REQUEST);
-      ctx.json(jsonProvider.objectToJSON(new BadRequest(e.getMessage())));
-    }
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
   }
 
-  protected String produceResultString(final BeaconBlock block) throws JsonProcessingException {
-    return jsonProvider.objectToJSON(new GetNewBlockResponse(block));
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final UInt64 slot = request.getPathParameter(SLOT_PARAMETER);
+    final BLSSignature randao = request.getQueryParameter(RANDAO_PARAMETER);
+    final Optional<Bytes32> graffiti = request.getOptionalQueryParameter(GRAFFITI_PARAMETER);
+
+    SafeFuture<Optional<BeaconBlock>> future =
+        provider.getUnsignedBeaconBlockAtSlot(slot, randao, graffiti);
+
+    request.respondAsync(
+        future.thenApply(
+            maybeBlock ->
+                maybeBlock
+                    .map(AsyncApiResponse::respondOk)
+                    .orElseThrow(ChainDataUnavailableException::new))); // TODO is this correct?
```

Comment:
this should be ok. the one thing is that this will generate a 204 response, so we need to add 
```
.response(SC_NO_CONTENT, "Data is unavailable because the chain has not yet reached genesis")
```
the message can probably go into a constant in RestApiConstants, and I think potentially the last one where we changed it to be unavailable should be changed to this way of doing things...